web/pdf_history.js

```diff
@@ -405,14 +450,22 @@ class PDFHistory {
 
     // This case corresponds to navigation backwards in the browser history.
     if (state.uid < this._currentUid && this._position && this._destination) {
-      if (this._destination.page &&
-          this._destination.page !== this._position.first &&
-          this._destination.page !== this._position.page) {
+      let shouldGoBack = false;
+
+      if (this._destination.temporary) {
+        // Always replace a previous *temporary* position, before moving back.
+        this._pushOrReplaceState(this._position);
```

Comment:
Should we force-replace here, since it *always* needs to happen?

Reply:
Doing so would break things *very* badly, since it would overwrite the *current* history entry.

Since we get where when the user goes back in the history, the current browser history entry (`window.history.state`) now points to the new destination we should be navigating to.
By force-replacing in this case, you'd get stuck in the *current* position in the document, with all sorts of breakage (history non-functional as so on).

This is quite complicated to wrap your head around at first (don't ask me how I know :-), so if you'd like a can try and explain in a bit more detail!